File: 3.EDD/5°Version/Lista/Java/Financiera.java
import java.util.Scanner;
import java.util.LinkedList;

public class Financiera {
    
    public static void main (String[] args){
        Scanner mex = new Scanner (System.in);
        LinkedList<Prestamo> lista = new LinkedList<>();

        int continuar =1;
        String eleccion;

        while(continuar != 0){
            Prestamo P1 = new Prestamo();

            System.out.println("\n \n");
            System.out.println("Datos del acreedor");
            System.out.println("\n");
            System.out.println("Ingresa los nombres: ");
            P1.setNombre(mex.nextLine());
            System.out.println("Ingresa los apellidos: ");
            P1.setApellido(mex.nextLine());
            System.out.println("Ingresa un numero telefonico: ");
            P1.setTelefono(mex.nextLine());
            System.out.println("Ingresa un correo electronico (Opcional): ");
            P1.setCorreo(mex.nextLine());

            System.out.println("Ingrese la cantidad a prestar: ");
            P1.setPrestamo(mex.nextDouble());
            System.out.println("Ingresar el porcentaje de interes: ");
            P1.setPorcentaje(mex.nextDouble());
            System.out.println("Ingresa la cantidad de a pagar: ");
            P1.setSemana(mex.nextDouble());
            mex.nextLine();

            System.out.println("[i] Agregar al inicio y [f] Agregar al final | Eleccion: ");
            eleccion = mex.nextLine();
            if(eleccion.equals("i")){
                lista.addFirst(P1);
            } else {
                lista.addLast(P1);
            }
            System.out.println("\n El prestamo a sido registrado");


          

            System.out.println("Para continuar oprima 1, para finalizar 0: ");
            continuar = mex.nextInt();
            mex.nextLine();
        }

        System.out.println("Extraer lista \n");
        while(!lista.isEmpty()){
            Prestamo iterador;
            System.out.println("[i] Extrae del inicio. [f] Extrae del final | Eleccion: ");
            eleccion = mex.nextLine();
            if(eleccion.equals("i")){
                iterador=lista.removeFirst();
            } else {
                iterador = lista.removeLast();
            }
            System.out.println(iterador.ToString());

        }

        mex.close();
    }
}

File: 3.EDD/5°Version/Lista/C++/Financiera.cpp
#include <iostream>
#include <list>
#include <cstdlib>
#include "Prestamo.h"
using namespace std;


int main(){
    system("color 09");
    list<Prestamo> coleccion;
    int continuar;

    string palabras, eleccion;
    double cantidad;
    Prestamo P1;

    do{
        // Datos del acreedor-------------------
        cout << "Ingresar datos del acreedor";
        cout << "\n \n";
        cout << "Ingresar nombres: ";
        getline(cin,palabras);
        P1.setNombre(palabras);
        cout << "Ingresar apellidos: ";
        getline(cin,palabras);
        P1.setApellido(palabras);
        cout << "Ingresar un numero telefonico: ";
        getline(cin,palabras);
        P1.setTelefono(palabras);
        cout << "Ingresar correo electronico (Opcional): ";
        getline(cin,palabras);
        P1.setCorreo(palabras);

        //Datos del pretamo-------------------------
        cout <<"\n \n";
        cout << "Ingresar la cantidad a prestar: ";
        cin >> cantidad;
        P1.setPrestamo(cantidad);
        cout << "Ingresar el porcentaje de interes (%): ";
        cin >> cantidad;
        P1.setPorcentaje(cantidad);
        cout << "Ingresar la cantidad de semanas a pagar: ";
        cin >> cantidad;
        P1.setSemana(cantidad);

        // Agregar a la lista
        cout << "[i] Agregar al inicio. [f] Agregar al final | elige: ";
        cin >> eleccion;
        if(eleccion == "i"){
            coleccion.push_front(P1);
        } else {
            coleccion.push_back(P1);
        }
        cout << "Prestamo registrado...... \n";
        cout << "Presiona 1 Para continuar y o para finalizar: ";
        cin >> continuar;
        cin.ignore();

        

    }while(continuar == 1);
    // Impresion lista----------------------------------------
        cout << "Extrayendo los elementos de la lista.\n";
        cout << "-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n";	
        while(!coleccion.empty()){
            Prestamo ejemplo;
            string eleccion;
            cout << "[i] Extrae del inicio. [f] Extrae del final. | Eleccion: ";
            cin >> eleccion;
            if(eleccion == "i"){
                ejemplo= coleccion.front();
                coleccion.pop_front();
            } else {
                ejemplo= coleccion.back();
                coleccion.pop_back();
            }
                
            cout << ejemplo.toString();
        }

    return 0;
}